src/key.rs

```diff
@@ -512,11 +509,13 @@ impl RsaPrivateKey {
 /// Check that the public key is well formed and has an exponent within acceptable bounds.
 #[inline]
 pub fn check_public(public_key: &impl PublicKeyParts) -> Result<()> {
-    if public_key.e() < &*MIN_PUB_EXPONENT {
+    let public_key: u64 = public_key.e().to_u64().ok_or(Error::PublicExponentTooLarge)?;
```

Comment:
IMO the `to_u64` is clear enough this doesn't need a `public_key: u64` type annotation

Reply:
Fixed.
Thank you for the review.